### ISSUE TITLE
BUG:Token type masking error in classifier

### ISSUE THREAD

**Issue by Liam-L2** [tyxqc/CoViPAL#3]
**Description**
In `modeling_qwen2_train1.py`, the classifier implementation uses a position-based incorrect mask to distinguish visual tokens from text tokens, which leads to incorrect prediction of token importance.

**Current Implementation (`modeling_qwen2_train1.py`, lines 620-624)**

```python
mask = torch.arange(classifier_weight.shape[-1]).unsqueeze(0).expand(classifier_weight.shape[0], -1) >= torch.tensor(self.classifier.image_token_len).unsqueeze(1)
classifier_weight[mask] = 0
```

**Issue Analysis**

1. **Incorrect token position assumption**
   The current code assumes:

   * First `image_token_len` positions = visual tokens ✗
   * Remaining positions = text tokens ✗

2. **Actual token sequence structure**
   According to `llava_arch.py` lines 474-495 (`prepare_inputs_labels_for_multimodal`), the actual sequence is:

   ```
   [text token, visual token, text token, visual token, text token, …]
   ```

   Visual tokens and text tokens are interleaved, not contiguous by position.

3. **Code evidence**
   `llava_arch.py` line 482:

   ```python
   cur_new_labels.append(torch.full((cur_image_features.shape[0],), IGNORE_INDEX, device=cur_labels.device, dtype=cur_labels.dtype))
   ```

   This indicates that visual tokens are marked with `IGNORE_INDEX (-100)` in the labels, providing the correct way to identify token types.
